inst/EpiModel 2.0 Introduction.rmd

```diff
@@ -0,0 +1,254 @@
+?EpiModel::dat_get_set
+```
+
+# Pre-EpiModel 2.0 Code Migration
+
+The following section will help EpiModel users update code written in the EpiModel 1.0 language to EpiModel 2.0.
+
+## Network_initialize replaces network.initialize
+
+New ```tergmLite``` version of network initilization function:
+
+```{r, echo = TRUE, eval = FALSE}
+nw <- network.initialize(num = 100)
+```
+
+is updated to
+
+```{r, echo = TRUE, eval = FALSE}
+nw <- network_initialize(num = 100)
+```
+
+Similarly, ```set.vertex.attribute``` and ```get.vertex.attribute``` have beenu updated as well:
+```{r echo = TRUE, eval = FALSE}
+nw <- set.vertex.attribute(nw, "group", rep(1:2, 50))
+group <- get.vertex.attribute(nw, "group")
+```
+
+is updated to
+
+```{r echo = TRUE, eval = FALSE}
+nw <- set_vertex_attribute(nw, "group", rep(1:2, 50))
+group <- get_vertex_attribute(nw, "group")
+```
+
+## Two-Group Network Setup
+
+For bipartite networks, i.e. a network in which the two groups do not have within group edge formation:
+
+```{r echo = TRUE, eval = FALSE}
+num1 <- num2 <- 50
+nw <- network.initialize(num = num1+num2, bipartite = num1)
+formation <- ~edges
+target.stats <- 30
+```
+
+```{r echo = TRUE, eval = FALSE}
+num1 <- num2 <- 50
+nw <- network_initialize(num = num1+num2)
+nw <- set_vertex_attribute(nw, "group", rep(1:2, c(num1, num2)))
+formation <- ~edges + nodematch("group")
+target.stats <- c(30, 0)
+```
+
+## Epidemic Type
+
+When ```control.net``` input parameter ```type``` is ```NULL```,
+
+```{r echo = TRUE, eval = FALSE}
+control <- control.net(type = NULL, nsims = 10, nsteps = 10,
+                       tergmLite = FALSE, save.network = FALSE, verbose = FALSE)
+```
+
+the end user must specify all base modules:
+
+```{r echo = TRUE, eval = FALSE}
+control <- control.net(type = NULL, nsims = 10, nsteps = 10,
+                       resim_net.FUN = resim_nets, infection.FUN = infection.net,
+                       recovery.FUN = recovery.net, prevalence.FUN = prevalence.net,
+                       departures.FUN = departures.net, arrivals.FUN = arrivals.net,
+                       tergmLite = FALSE, save.network = FALSE, verbose = FALSE)
+```
+
+## Depend to resimulate.network
+
+The ```control.net``` parameter used to resimulate the network at each time has been renamed from depend
+
+```{r echo = TRUE, eval = FALSE}
+control <- control.net(type = "SI", nsims = 10, nsteps = 10, depend = TRUE)
+```
+
+to resimulate.network
+
+```{r echo = TRUE, eval = FALSE}
+control <- control.net(type = "SI", nsims = 10, nsteps = 10, resimulate.network = TRUE)
+```
+
+## Mode to Group Parameters
+
+Following the move from mode to group language, mode syntax has been updated to group syntax. The following:
+
+```{r echo = TRUE, eval = FALSE}
+init <- init.net(i.num = 50, i.num.m2 = 50)
+param <- param.net(inf.prob = 0.1, inf.prob.m2 = 0.1,
+                   rec.rate = 0.02, rec.rate.m2 = 0.02,
+                   act.rate = 5)
+```
+
+has been updated to
```

Comment:
"should be changed to" You are speaking to a user who has one code base that requires updating.